startupshutdown/startupshutdown.go

```diff
@@ -1,73 +1,81 @@
 package startupshutdown
 
 import (
-	"log"
-	"os"
-	"os/exec"
+	"fmt"
 	"strings"
 
+	"github.com/google/uuid"
 	"github.com/mholt/caddy"
+	"github.com/mholt/caddy/onevent/hook"
 )
 
 func init() {
 	caddy.RegisterPlugin("startup", caddy.Plugin{Action: Startup})
 	caddy.RegisterPlugin("shutdown", caddy.Plugin{Action: Shutdown})
 }
 
-// Startup registers a startup callback to execute during server start.
+// Startup is an "alias" for on startup.
 func Startup(c *caddy.Controller) error {
-	return registerCallback(c, c.OnFirstStartup)
+	config, err := onParse(c, caddy.InstanceStartupEvent)
+	if err != nil {
+		return c.ArgErr()
+	}
+
+	// Register Event Hooks.
+	for _, cfg := range config {
+		caddy.RegisterEventHook("on-"+cfg.ID, cfg.Hook)
+	}
+
+	fmt.Println("NOTICE: Startup directives will be removed in a later version. Please migrate to 'on startup'")
+
+	return nil
 }
 
-// Shutdown registers a shutdown callback to execute during server stop.
+// Shutdown is an "alias" for on shutdown.
 func Shutdown(c *caddy.Controller) error {
-	return registerCallback(c, c.OnFinalShutdown)
+	config, err := onParse(c, caddy.ShutdownEvent)
+	if err != nil {
+		return c.ArgErr()
+	}
+
+	// Register Event Hooks.
+	for _, cfg := range config {
+		caddy.RegisterEventHook("on-"+cfg.ID, cfg.Hook)
+	}
+
+	fmt.Println("NOTICE: Shutdown directives will be removed in a later version. Please migrate to 'on shutdown'")
```

Comment:
Change text slightly to use directive instead of directives

    "NOTICE: Shutdown directive will be removed in a later version. Please migrate to 'on shutdown'")